### ISSUE TITLE
Can't install latest matplotlib from conda-forge

### ISSUE THREAD

**Issue by JanisGailis** [conda-forge/matplotlib-feedstock#131]
Hi!

I hope this is the correct place where to file this issue.

Our build broke today and it seems that the culprit is not being able to install the latest Matplotlib (2.0.2, py36_2) from conda-forge anymore. This then results in Matplotlib 2.0.2 being pulled from the 'default' conda repositories, which has issues of its own and breaks the build.

Running:
```shell
$ conda create -c conda-forge --override-channels -n cate_test matplotlib
```
Wants to install an old Matplotlib package and pull in Qt4:
```shell
The following NEW packages will be INSTALLED:

    ca-certificates: 2017.7.27.1-0      conda-forge
    cairo:           1.14.6-4           conda-forge
    certifi:         2017.7.27.1-py36_0 conda-forge
    cycler:          0.10.0-py36_0      conda-forge
    fontconfig:      2.12.1-4           conda-forge
    freetype:        2.7-1              conda-forge
    gettext:         0.19.7-1           conda-forge
    glib:            2.51.4-0           conda-forge
    graphite2:       1.3.9-0            conda-forge
    harfbuzz:        1.4.3-0            conda-forge
    icu:             58.1-1             conda-forge
    jpeg:            9b-0               conda-forge
    libffi:          3.2.1-3            conda-forge
    libiconv:        1.14-4             conda-forge
    libpng:          1.6.28-0           conda-forge
    libtiff:         4.0.6-7            conda-forge
    libxml2:         2.9.5-0            conda-forge
    matplotlib:      2.0.0-np112py36_2  conda-forge
    mkl:             2017.0.3-0                    
    ncurses:         5.9-10             conda-forge
    numpy:           1.12.1-py36_0                 
    openssl:         1.0.2l-0           conda-forge
    pango:           1.40.4-0           conda-forge
    pcre:            8.39-0             conda-forge
    pip:             9.0.1-py36_0       conda-forge
    pixman:          0.34.0-0           conda-forge
    pyparsing:       2.2.0-py36_0       conda-forge
    pyqt:            4.11.4-py36_2      conda-forge
    python:          3.6.2-0            conda-forge
    python-dateutil: 2.6.1-py36_0       conda-forge
    pytz:            2017.2-py36_0      conda-forge
    qt:              4.8.7-7            conda-forge
    readline:        6.2-0              conda-forge
    setuptools:      36.3.0-py36_0      conda-forge
    sip:             4.18-py36_1        conda-forge
    six:             1.11.0-py36_1      conda-forge
    sqlite:          3.13.0-1           conda-forge
    tk:              8.5.19-2           conda-forge
    tornado:         4.5.2-py36_0       conda-forge
    wheel:           0.30.0-py_1        conda-forge
    xz:              5.2.3-0            conda-forge
    zlib:            1.2.11-0           conda-forge

Proceed ([y]/n)? 
```

Trying to set it explicitly to the latest version results in Matplotlib being pulled from 'default':
```shell
$ conda create -c conda-forge --override-channels -n cate_test matplotlib=2.0.2
```
```shell
The following NEW packages will be INSTALLED:

    ca-certificates:  2017.7.27.1-0      conda-forge
    certifi:          2017.7.27.1-py36_0 conda-forge
    cycler:           0.10.0-py36_0      conda-forge
    dbus:             1.10.22-0          conda-forge
    expat:            2.2.1-0            conda-forge
    fontconfig:       2.12.1-3                      
    freetype:         2.5.5-2                       
    gettext:          0.19.7-1           conda-forge
    glib:             2.51.4-0           conda-forge
    gst-plugins-base: 1.8.0-0            conda-forge
    gstreamer:        1.8.0-2            conda-forge
    icu:              54.1-0                        
    jpeg:             9b-0               conda-forge
    libffi:           3.2.1-3            conda-forge
    libgcc:           5.2.0-0                       
    libiconv:         1.14-4             conda-forge
    libpng:           1.6.28-0           conda-forge
    libxcb:           1.12-1             conda-forge
    libxml2:          2.9.4-0                       
    matplotlib:       2.0.2-np113py36_0             
    mkl:              2017.0.3-0                    
    ncurses:          5.9-10             conda-forge
    numpy:            1.13.1-py36_0                 
    openssl:          1.0.2l-0           conda-forge
    pcre:             8.39-0             conda-forge
    pip:              9.0.1-py36_0       conda-forge
    pyparsing:        2.2.0-py36_0       conda-forge
    pyqt:             5.6.0-py36_4       conda-forge
    python:           3.6.2-0            conda-forge
    python-dateutil:  2.6.1-py36_0       conda-forge
    pytz:             2017.2-py36_0      conda-forge
    qt:               5.6.2-5                       
    readline:         6.2-0              conda-forge
    setuptools:       36.3.0-py36_0      conda-forge
    sip:              4.18-py36_1        conda-forge
    six:              1.11.0-py36_1      conda-forge
    sqlite:           3.13.0-1           conda-forge
    tk:               8.5.19-2           conda-forge
    wheel:            0.30.0-py_1        conda-forge
    xorg-libxau:      1.0.8-3            conda-forge
    xorg-libxdmcp:    1.1.2-3            conda-forge
    xz:               5.2.3-0            conda-forge
    zlib:             1.2.11-0           conda-forge

Proceed ([y]/n)? 
```
Trying to set explicitly the latest build that we know worked, fails to produce an environment:
```shell
$ conda create -c conda-forge --override-channels -n cate_test matplotlib=2.0.2=py36_2
```
```shell
UnsatisfiableError: The following specifications were found to be in conflict:
  - matplotlib 2.0.2 py36_2
Use "conda info <package>" to see the dependencies for each package.
```
It's probably caused by some change upstream in another package, as the last commit in this repo was a few days ago, while this only happened today. Still, Matplotlib build seems to be breaking.

Please let us know if there's something trivial we're missing!

**Comment by abuckenheimer** [comment]
Were seeing the same issue, I don't know why but there are two different packages with two different sets of dependencies described in the default channel vs the conda-forge channel

```
$conda info matplotlib=2.0.2=np113py36_0
Fetching package metadata .............

matplotlib 2.0.2 np113py36_0
----------------------------
file name   : matplotlib-2.0.2-np113py36_0.tar.bz2
name        : matplotlib
version     : 2.0.2
build string: np113py36_0
build number: 0
channel     : defaults
size        : 10.9 MB
arch        : x86_64
date        : 2017-06-07
license     : PSF-based
license_family: PSF
md5         : bbc68dd74f0115c0604c4d093454c0b7
noarch      : None
platform    : linux
url         : https://repo.continuum.io/pkgs/free/linux-64/matplotlib-2.0.2-np113py36_0.tar.bz2
dependencies:
    cycler >=0.10.0
    freetype 2.5.*
    libpng >=1.6.27,<1.7
    numpy 1.13*
    pyparsing
    pyqt 5.*
    python 3.6*
    python-dateutil
    pytz

matplotlib 2.0.2 np113py36_0
----------------------------
file name   : matplotlib-2.0.2-np113py36_0.tar.bz2
name        : matplotlib
version     : 2.0.2
build string: np113py36_0
build number: 0
channel     : conda-forge
size        : 9.4 MB
arch        : x86_64
has_prefix  : False
license     : PSF-based
license_family: PSF
md5         : 4fc25fef0c6f21274af1b5c4894a43af
noarch      : None
platform    : linux
requires    : ()
subdir      : linux-64
url         : https://conda.anaconda.org/conda-forge/linux-64/matplotlib-2.0.2-np113py36_0.tar.bz2
dependencies:
    cycler >=0.10
    freetype 2.7|2.7.*
    icu 58.*
    libpng >=1.6.28,<1.7
    numpy 1.13*
    pyparsing
    pyqt 5.6.*
    python 3.6*
    python-dateutil
    pytz
    setuptools
    tk 8.5.*
    tornado
```

**Comment by jakirkham** [comment]
My guess is this is because we are shy a `qt` 5 package on Linux. We are currently working on building one and getting it on conda-forge.

**Comment by dopplershift** [comment]
@jakirkham I must be missing something? I see qt 5 for linux in conda-forge:
```
>conda search --override-channels --channel conda-forge --platform linux-64 --spec qt=5              [master●●]
Fetching package metadata ...........
qt                           5.6.0                         0  defaults        
                             5.6.0                         1  defaults        
                             5.6.2                         0  defaults        
                             5.6.2                         1  defaults        
                             5.6.2                         2  defaults        
                             5.6.2                         3  defaults        
                             5.6.2                         4  defaults        
                             5.6.2                         5  defaults        
                             5.6.2                         3  conda-forge     
```

**Comment by jakirkham** [comment]
Appears that was changed ~1hr ago. So it was the case, but no longer. Given this change, does the issue still exist?

**Comment by ocefpaf** [comment]
@abuckenheimer we are in the process of testing the use of `qt` from the default channel instead of conda-forge. That is due to a limitation in our CIs where we cannot build them. As long as we can keep the dependencies in sync I believe that is the best option we have at the moment.

TL;DR something like this may happen again but it should go away in 10 hours tops. If not then it is a real issue :smile:

**Comment by JanisGailis** [comment]
10 hours later, our build seems to be working fine! I'm going to close this, thanks for letting us know what's going on!

**Comment by ocefpaf** [comment]
Thanks @JanisGailis. Heads up, I will be experimenting again this weekend. Things may be unstable for a few hours.
